src/devices/Mpu9250/README.md

```diff
@@ -0,0 +1,180 @@
+var mag = mpu9250.CalibrateMagnetometer();
+Console.WriteLine($"Bias:");
+Console.WriteLine($"Mag X = {mpu9250.MagnometerBias.X}");
+Console.WriteLine($"Mag Y = {mpu9250.MagnometerBias.Y}");
+Console.WriteLine($"Mag Z = {mpu9250.MagnometerBias.Z}");
+```
+
+## Units
+
+Al axis are oriented this way:
+            +Z   +Y
+          \  |  /
+           \ | /
+            \|/
+            /|\
+           / | \
+          /  |  \
+                 +X
+
+### Gyroscope
+
+The unit used for the gyroscope are degree per seconds.
+
+### Accelerometer
+
+The unit used for the accelerometer is G.
+
+### Magnetometer
+
+The unit used for the magnetometer is µTesla.
+
+### Temperature
+
+The Temperature is a normalized Units.Temperature which can provide Celsius, Kelvin or Fahrenheit degrees.
+
+## Measurement modes
+
+The MPU9250 offers a large variety of measurement modes. They can be changed and adjusted thru the properties like:
+
+* ```MagnetometerMeasurementMode``` to adjust the type of measurement for the magnetometer
+* ```MagnetometerOutputBitMode``` to select between 14 and 16 bits precision of the magnetometer
+* ```AccelerometerRange``` to adjust the range of the accelerometer between 2, 4, 8 or 16 G
+* ```AccelerometerBandwidth``` to adjust the frequency of measurement from 5 Hz to 1130 Hz
+* ```GyroscopeRange``` to adjust the range of the gyroscope from 250, 500, 1000 and 2000 degrees per second
+* ```GyroscopeBandwidth``` to adjust the frequency of measurement from 5 Hz to 8800 Hz
+* ```SampleRateDivider``` allows you to reduce the number of samples for the gyroscope and the accelerometer. This feature is only available for some of the bandwidth modes.
+* ```DisableModes``` allows you to disable any of the gyroscope and accelerometer axis
+
+### Wake on motion
+
+A unique ```SetWakeOnMotion``` mode is available. It puts the MPU9250 in a low consumption, low measurement rate mode and trigger an interruption on the INT pin. 
+
+```csharp
+mpu9250.SetWakeOnMotion(300, AccelerometerLowPowerFrequency.Frequency0Dot24Hz);
+// You'll need to attach the INT pin to a GPIO and read the level. Once going up, you have 
+// some data and the sensor is awake
+// In order to simulate this without a GPIO pin, you will see that the refresh rate is very low
+// Setup here at 0.24Hz which means, about every 4 seconds
+
+while (!Console.KeyAvailable)
+{
+    Console.CursorTop = 0;
+    var acc = mpu9250.Accelerometer;
+    Console.WriteLine($"Acc X = {acc.X}          ");
+    Console.WriteLine($"Acc Y = {acc.Y}          ");
+    Console.WriteLine($"Acc Z = {acc.Z}          ");
+    Thread.Sleep(100);
+}
+```
+
+### FIFO mode
+
+The Fifo mode allows you to get the data by batch. You can select the mode thru ```FifoModes```, then read the ```FifoCount``` property. You can then read the data thru ```ReadFifo``` Make sure you'll size the ```Span<byte>``` with ```FifoCount``` length.
```

Comment:
I feel like this should be either hidden or exposed differently (i.e. ReadBatch)

Reply:
The FIFO mode is a mode itself. Depending on the choice you are making, the results can be very different, that's the reason it is exposed like this.